internal/testtypes/builders.go

```diff
@@ -523,3 +544,20 @@ func setNestedRef(obj, refObj *unstructured.Unstructured, fields ...string) erro
 	}
 	return unstructured.SetNestedField(obj.UnstructuredContent(), ref, fields...)
 }
+
+// setSpecFields sets fields in an unstructured object from a map.
+func setSpecFields(obj *unstructured.Unstructured, fields map[string]interface{}) {
+	for k, v := range fields {
+		fieldParts := strings.Split(k, ".")
+		if len(fieldParts[0]) == 0 {
+			return
```

Comment:
```suggestion
			continue
```
Not entirely sure if return or continue is intended

Reply:
Another option is to also panic (like below) as I think it's never intended to have a test with an basically invalid path

Reply:
You might also want to check for the length of fieldParts, because otherwise it might panic here (although that's maybe fine as we want to panic here anyway in that case?)

Reply:
Should be fixed now - I'm looking for the length and panicking if it's zero instead of this check.